docs/misc/ConfigDB_state.md

```diff
@@ -0,0 +1,84 @@
+Automatic transition from NotConfigured is done by Discovery (see below)
+
+* NotConfigured -> NotConfigured    Keep going.
+* NotConfigured -> SingleNode       Forced by by creating a new DB in /etc/vmware/vmdkops (AuthenticationDataManager.new_db(). When discovered, we just connect to the DB, log an INFO message and keep going.
+* NotConfigured -> MutliNode         Forced by creating the DB on a shred VMFS volume and creating symlink in /etc/vmware/vmdkops. When discovered, we just connect to the DB, log an INFO message and keep going. (same as in NotConfigured->MutliNode)
+* NotConfigured -> BrokenLink       ERROR - throw exception
+
+### BrokenLink handling
+BrokenLink is an error condition. We do not know what was the admin intent so we fail all Docker volume requests until the situation is resolved by the admin using `config rm --local` command
+
+* BrokenLink    ->  anything forced by admin.
+
+# From SingleNode
+
+Transition from SingleNode means the admin live-reconfigured the Config DB.
+On the node where the operation happened, the admin CLI will refreh the vDVS service which will find out the new state from the DB.
+
+On other nodes, the discovery will find out about the change to `MultiNode` only, and will issue a warning to the admin  - since we do not know what was the intent.
+
+* SingleNode    -> Single Node      noop
+* SingleNode    -> NotConfigured    happend on `config -rm --local` only. Restart the service.
+* SingeNode     -> MutliNode        can happen on discovery. WARNING - Config changed. No action  we do not know what was the intent.
+* SingeNode     -> BrokenLink       ERROR - Throw exception
+
+## From MultiNode
+
+To transition from MultiNode, either the admin has to issue `config --rm` or there should be a FS or DB issue (i.e. the symink was accidentaly removed)
+
+* MutliNode      -> MutliNode       noop
+* MultiNode      -> NotConfigured   restart the service (on discovery or on CLI)
+* MutliNode      -> SingleNode      same
+* MutliNode      -> BrokenLink      ERROR - Throw exception
+
+## Discovery
+
+Discovery goal is to sync `vmdk_ops.py` Config DB state to reflect changes to the state done on a different ESXi node.
+
+**The key use case** is initial configuration, where all nodes start with NotConfigured. On one of the nodes, admin (manually or via automation) configures a shared Config DB on a datastore visible to all nodes. The on all nodes  discovery kicks in, finds out there is a fresh Config DB on a shared store, and configures the node to use this DB. This way Admin only need to do configuration on one node and the rest pick it up . See Issue https://github.com/vmware/docker-volume-vsphere/issues/1086 for details.
+
+The only state transition supported by Discovery is NotConfigured -> MultiNode.
+*  Note - once defined, the DB becomes a single point of failure, and the only way to remove it is to run local CLI `config rm --local` on each ESX
```

Comment:
I'm a little bit confused here - is "config rm --local" supposed to remove the local DB only, or it also supports to remove shared DB? Also see my comments below.